### PR TITLE
add spanmetrics connector and count connector

### PR DESCRIPTION
**Pull request by greatestusername-splunk**
Original issue: https://github.com/signalfx/splunk-otel-collector/issues/3298

This PR adds:
- `spanmetrics` connector: Currently the processor is included but is more than a little fiddly to setup. The connector streamlines this.
- `count` connector: Count connector allows creating metrics from logs (and other types like spans, traces, etc). 
- Adds these connectors to the docs

Examples would be provided in the next PR

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,6 +15,7 @@ require (
 	github.com/hashicorp/vault/api v1.9.2
 	github.com/jaegertracing/jaeger v1.41.0
 	github.com/knadh/koanf v1.5.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector v0.80.0
```
Comment (rmfitzpatrick) [inline go.mod:18]: To prevent moving the deps forward, do you mind using 0.79.0 for now? We have an automated job that will address them.

Comment (greatestusername-splunk) [reply go.mod:18]: Can do!

File: docs/components.md [hunk 2]
```diff
@@ -116,9 +116,11 @@ The distribution offers support for the following components.
 
 <div>
 
-| Connectors                                                                                                           | Stability        |
-|:---------------------------------------------------------------------------------------------------------------------|:-----------------|
-| [forward](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector/forwardconnector)            | [alpha]          |
+| Connectors                                                                                                                 | Stability        |
+|:---------------------------------------------------------------------------------------------------------------------------|:-----------------|
+| [forward](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector/forwardconnector)                  | [alpha]          |
```
[the remark this answers is not in the document]
Comment (greatestusername-splunk) [reply docs/components.md:121]: Will do!

**Comment by greatestusername-splunk** [comment]
Review comments are addressed and checks are passing. Let me know if you notice anything else you'd like changed up.

Thanks again!!

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**